aws/gpu_tree_shap/gpu_tree_shap.ipynb

```diff
@@ -0,0 +1,479 @@
+    "prefix = \"sagemaker/DEMO-xgboost-inference-script-mode\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 1. Training the XGBoost model\n",
+    "\n",
+    "SageMaker can now run an XGboost script using the [XGBoost estimator](https://docs.aws.amazon.com/sagemaker/latest/dg/xgboost.html). A typical training script loads data from the input channels, configures training with hyperparameters, trains a model, and saves a model to `model_dir` so that it can be hosted later. In this notebook, we use the training script **train.py**.\n",
+    "\n",
+    "After setting training parameters, we kick off training. \n",
+    "\n",
+    "To run our training script on SageMaker, we construct a `sagemaker.xgboost.estimator.XGBoost` estimator, which accepts several constructor arguments:\n",
+    "\n",
+    "* __entry_point__: The path to the Python script SageMaker runs for training and prediction.\n",
+    "* __role__: Role ARN\n",
+    "* __framework_version__: SageMaker XGBoost version you want to use for executing your model training code, e.g., `0.90-1`, `0.90-2`, `1.0-1`, or `1.3-1`. We must use `1.3-1` for GPU accelerated SHAP values.\n",
+    "* __train_instance_type__ *(optional)*: The type of SageMaker instances for training. __Note__: Because Scikit-learn does not natively support GPU training, Sagemaker Scikit-learn does not currently support training on GPU instance types.\n",
+    "* __sagemaker_session__ *(optional)*: The session used to train on Sagemaker.\n",
+    "* __hyperparameters__ *(optional)*: A dictionary passed to the train function as hyperparameters. For the XGBoost estimator, the list of possible hyperparameters can be found [here](https://docs.aws.amazon.com/sagemaker/latest/dg/xgboost_hyperparameters.html). "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "from sagemaker.inputs import TrainingInput\n",
+    "from sagemaker.xgboost.estimator import XGBoost\n",
+    "\n",
+    "job_name = \"DEMO-xgboost-inference-script-mode-\" + time.strftime(\"%Y-%m-%d-%H-%M-%S\", time.gmtime())\n",
+    "print(\"Training job\", job_name)\n",
+    "\n",
+    "hyperparameters = {\n",
+    "    \"max_depth\": \"6\",\n",
+    "    \"eta\": \"0.3\",\n",
+    "    \"gamma\": \"0\",\n",
+    "    \"min_child_weight\": \"1\",\n",
+    "    \"subsample\": \"1\",\n",
+    "    \"objective\": \"reg:squarederror\",\n",
+    "    \"num_round\": \"500\",\n",
+    "    \"verbosity\": \"1\",\n",
+    "    \"sklearn_dataset\": \"sklearn.datasets.fetch_california_housing()\",\n",
+    "    # using default \"tree_method\": \"gpu_hist\" and \"predictor\": \"gpu_predictor\"\n",
+    "    # \"tree_method\": \"hist\", \"predictor\": \"cpu_predictor\",  # for CPU version\n",
+    "}\n",
+    "\n",
+    "instance_type = \"ml.g4dn.xlarge\"  # \"ml.c5.xlarge\"\n",
```

Comment:
Mention that this is a GPU instance and that they can enable the CPU instance type for the CPU run.